smart_open/compression.py

```diff
@@ -15,6 +15,14 @@
 _COMPRESSOR_REGISTRY = {}
 
 
+NO_COMPRESSION = 'none'
+INFER_FROM_EXTENSION = 'extension'
+
+
+def get_supported_compression_types():
```

Comment:
Is this an internal method, or something that our users will see? In the case of the latter, please add a docstring. Without looking at the actual implementation, it's unclear how this differs from get_supported_extensions.

Reply:
This would be public; I'll add a docstring. The idea here is that, because the names suggested in the design discussion were 1:1 with the supported extensions, there's no reason to duplicate the list, if they can be generated automatically. On the other hand, a dynamic listing leaves end users without a way to explicitly enumerate the options from constants. This is meant to by dynamic, in that, if a user registers another extension handler for compression, they will see another valid option to pass via the `compression` parameter, and it will be handled seamlessly.

- [ ] add a docstring